src/hooks/useFilterPossiblyMaliciousPositions.ts

```diff
@@ -0,0 +1,56 @@
+import { Token } from '@uniswap/sdk-core'
+import { useMemo } from 'react'
+import { PositionDetails } from 'types/position'
+import { hasURL } from 'utils/urlChecks'
+
+import { useDefaultActiveTokens } from './Tokens'
+import { useTokenContractsConstant } from './useTokenContractsConstant'
+
+function getUniqueAddressesFromPositions(positions: PositionDetails[]): string[] {
+  return Array.from(
+    new Set(positions.reduce<string[]>((acc, position) => acc.concat(position.token0, position.token1), []))
+  )
+}
+
+export function useFilterPossiblyMaliciousPositions(positions: PositionDetails[]): PositionDetails[] {
+  const activeTokensList = useDefaultActiveTokens()
+
+  const nonListPositionTokenAddresses = useMemo(
+    () => getUniqueAddressesFromPositions(positions).filter((address) => !activeTokensList[address]),
+    [positions, activeTokensList]
+  )
+
+  const symbolCallStates = useTokenContractsConstant(nonListPositionTokenAddresses, 'symbol')
+
+  const addressesToSymbol: Record<string, string> = useMemo(() => {
+    const result: Record<string, string> = {}
+    for (let i = 0; i < nonListPositionTokenAddresses.length; i++) {
```

Comment:
nit: prefer `for... of` instead

Reply:
If both elements are an array, could you do something like:
```
Object.fromEntries(nonListPositionTokenAddresses, symbolCallStates)
```
but you would need to make sure `symbolCallStates` is previously mapped to flat array, instead of having a nested `result`.